web/packages/extension/src/background.ts

```diff
@@ -8,7 +8,138 @@ async function contentScriptRegistered() {
     return matchingScripts?.length > 0;
 }
 
+async function isHeaderConditionSupported() {
```

Comment:
`// Copied from https://github.com/w3c/webextensions/issues/638#issuecomment-2181124486` would be appropriate.